OpenEdXMobile/src/main/java/org/edx/mobile/view/adapters/TranscriptAdapter.java

```diff
@@ -0,0 +1,58 @@
+package org.edx.mobile.view.adapters;
+
+import android.content.Context;
+import android.support.annotation.ColorInt;
+import android.support.v4.content.ContextCompat;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.TextView;
+
+import com.google.inject.Inject;
+
+import org.edx.mobile.R;
+import org.edx.mobile.core.IEdxEnvironment;
+
+import subtitleFile.Caption;
+
+public class TranscriptAdapter extends BaseListAdapter<Caption> {
+    @ColorInt
+    private final int edx_brand_gray_base = ContextCompat.getColor(getContext(), R.color.edx_brand_gray_base);
+    @ColorInt
+    private final int edx_brand_primary_base = ContextCompat.getColor(getContext(), R.color.edx_brand_primary_base);
+
+    @Inject
+    public TranscriptAdapter(Context context, IEdxEnvironment environment) {
+        super(context, R.layout.row_transcript_item, environment);
+    }
+
+    @Override
+    public void render(BaseViewHolder tag, Caption model) {
+        ViewHolder viewHolder = (ViewHolder) tag;
+        String captionText = model.content;
+        if (captionText.endsWith("<br />")) {
+            captionText = captionText.substring(0, captionText.length() - 6);
+        }
+        viewHolder.transcriptTv.setText(captionText);
+        final int position = getPosition(model);
```

Comment:
The alternative that you have given will also allocate the same memory. It will just limit the scope of the new allocation.

But since, position is something that might get used in the future, so, its safe to keep it as is.